frameworks/FSharp/frank/src/App/Program.fs

```diff
@@ -0,0 +1,88 @@
+module App.App
+
+open System
+open System.IO
+open System.Text
+open System.Text.Json.Serialization
+open System.Text.Json
+open System.Threading.Tasks
+open Microsoft.AspNetCore.Hosting
+open Microsoft.AspNetCore.Http
+open Microsoft.Extensions.Logging
+open Dapper
+open Giraffe
+open Frank.Builder
+open FSharp.Control.Tasks
+open Npgsql
+open Models
+
+let inline contentLength x = new Nullable<int64> ( int64 x )
+
+let json' data : HttpContext -> Task =
+    let options = JsonSerializerOptions()
+    options.Converters.Add(JsonFSharpConverter())
+    fun ctx ->
+        ctx.Response.ContentType <- "application/json"
+        ctx.Response.StatusCode <- 200
+        JsonSerializer.SerializeAsync(ctx.Response.Body, data)
+
+let text' (msg:string): HttpContext -> Task =
+    let bytes = Encoding.UTF8.GetBytes(msg)
+    fun ctx ->
+        ctx.Response.ContentLength <- contentLength bytes.Length
+        ctx.Response.ContentType <- "text/plain"
+        ctx.Response.StatusCode <- 200
+        ctx.Response.Body.WriteAsync(bytes, 0, bytes.Length)
+
+// Pulled from Giraffe example
+let fortunes' = 
+    let extra = { id = 0; message = "Additional fortune added at request time." }
+    fun next (ctx: HttpContext) ->
+        let conn = new NpgsqlConnection(ConnectionString)
+        ctx.Response.RegisterForDispose conn
+        task {
+            let! data = conn.QueryAsync<Fortune>("SELECT id, message FROM fortune")
+
+            let fortunes = 
+                let xs = data.AsList()
+                xs.Add extra
+                xs.Sort FortuneComparer
+                xs
+
+            return! htmlView (HtmlViews.fortunes fortunes) next ctx
+        }
+
+// Resources
+
+let plaintext = 
+    resource "/plaintext" {
+        name "Plain text"
+        get (text' "Hello, World!")
+    }
+
+let json =
+    resource "/json" {
+        name "JSON"
+        get (json' struct {|message="Hello, World!"|})
```

Comment:
Sorry to be a pain here, but this `struct` creation still needs to be moved into the function at line 24. My understanding of what is happening here is that this struct is only being created once and then being serialized on each request. It also needs to be instantiated on each request.

https://github.com/TechEmpower/FrameworkBenchmarks/wiki/Project-Information-Framework-Tests-Overview#json-serialization

> i. For each request, an object mapping the key message to Hello, World! must be instantiated.

Reply:
No worries. Sorry that I misunderstood.